packages/ui-color-picker/src/ColorMixer/__tests__/ColorMixer.test.tsx

```diff
@@ -128,4 +146,1591 @@ describe('<ColorMixer />', () => {
+    })
+  })
+
+  describe('hue slider', () => {
+    it('should not call onChange when receive `tab` key press', async () => {
+      const onChange = stub()
+      await mount(
+        <ColorMixer
+          withAlpha
+          {...testInputLabels}
+          {...testScreenReaderLabels}
+          onChange={onChange}
+        />
+      )
+
+      const component = await ColorMixerLocator.find()
+      const hueSlider = await component.findColorSlider()
+      await hueSlider.keyDown('tab')
+      expect(onChange).to.not.have.been.called()
+    })
+
+    it('onChange should not be call with click event when component is disable', async () => {
+      const onChange = stub()
+      await mount(
+        <ColorMixer
+          disabled
+          {...testInputLabels}
+          {...testScreenReaderLabels}
+          onChange={onChange}
+        />
+      )
+      const component = await ColorMixerLocator.find()
+      const hueSlider = await component.findColorSlider()
+      const x = hueSlider.getBoundingClientRect().x
+      await hueSlider.mouseDown({ clientX: x + 10 })
+      expect(onChange).to.have.not.been.called()
+    })
+
+    describe('hue value change when receive a mouse click event', async () => {
+      it('click at the middle of the slider', async () => {
+        const onChange = stub()
+        await mount(
+          <ColorMixer
+            {...testInputLabels}
+            {...testScreenReaderLabels}
+            onChange={onChange}
+          />
+        )
+        const component = await ColorMixerLocator.find()
+        const hueSlider = await component.findColorSlider()
+        const rect = hueSlider.getBoundingClientRect()
+        await hueSlider.mouseDown({ clientX: rect.x + rect.width / 2 })
+        expect(onChange).to.have.been.called()
+      })
+      it('click at the end of the slider', async () => {
+        const onChange = stub()
+        await mount(
+          <ColorMixer
+            {...testInputLabels}
+            {...testScreenReaderLabels}
+            onChange={onChange}
+          />
+        )
+        const component = await ColorMixerLocator.find()
+        const hueSlider = await component.findColorSlider()
+        const rect = hueSlider.getBoundingClientRect()
+        await hueSlider.mouseDown({ clientX: rect.x + rect.width })
+        expect(onChange).to.have.been.called()
+      })
+
+      it('click at the begin of the slider', async () => {
+        const onChange = stub()
+        await mount(
+          <ColorMixer
+            value="#00FF00"
+            {...testInputLabels}
+            {...testScreenReaderLabels}
+            onChange={onChange}
+          />
+        )
+        const component = await ColorMixerLocator.find()
+        const hueSlider = await component.findColorSlider()
+        const rect = hueSlider.getBoundingClientRect()
+        await hueSlider.mouseDown({ clientX: rect.x })
+        expect(onChange).to.have.been.calledTwice()
```

Comment:
Why is this called twice? Could this be a bug?

Reply:
Because as the default, the indicator was at the beginning of the slider then I had to move the indicator to somewhere in the middle of the slider by passing them a random value, then I click to move it back beginning so it will change the color twice, once when I pass the `props.value` and once when I click.